--- a/lib/fribidi-bidi.c
+++ b/lib/fribidi-bidi.c
@@ -558,12 +558,12 @@
           else if (valid_isolate_count==0 && FRIBIDI_IS_LETTER (RL_TYPE (pp)))
             {
               base_level = FRIBIDI_DIR_TO_LEVEL (RL_TYPE (pp));
-              *pbase_dir = FRIBIDI_LEVEL_TO_DIR (base_level);
+              *pbase_dir = (FriBidiParType) FRIBIDI_LEVEL_TO_DIR (base_level);
               break;
             }
         }
     }
-  base_dir = FRIBIDI_LEVEL_TO_DIR (base_level);
+  base_dir = (FriBidiParType) FRIBIDI_LEVEL_TO_DIR (base_level);
   DBG2 ("  base level : %c", fribidi_char_from_level (base_level));
   DBG2 ("  base dir   : %s", fribidi_get_bidi_type_name (base_dir));
 
@@ -1476,7 +1476,7 @@
 	      }
 	    p->pos = j + 1;
 	    p->len = pos - j;
-	    p->type = base_dir;
+	    p->type = (FriBidiCharType) base_dir;
 	    p->level = base_level;
 	    move_node_before (p, q);
 	    q = p;
