--- a/KeyStorage.cpp
+++ b/KeyStorage.cpp
@@ -335,13 +335,6 @@
 
     auto blob_file = dir + "/" + kFn_keymaster_key_blob;
     LOG(INFO) << "reading blob_file: " << blob_file;
-    std::string blob_dir(kFn_keymaster_key_blob);
-    std::string temp_dir = "/tmp/" + blob_dir + "/";
-    if (TEMP_FAILURE_RETRY(mkdir(temp_dir.c_str(), 0700)) == -1) {
-        PLOG(ERROR) << "key mkdir " << temp_dir;
-    }
-    auto upgraded_blob_file = temp_dir + kFn_keymaster_key_blob;
-    // auto upgraded_blob_file = dir + "/" + kFn_keymaster_key_blob_upgraded;
     std::lock_guard<std::mutex> lock(key_upgrade_lock);
 
     std::string blob;
@@ -360,7 +353,15 @@
     if (!opHandle) return opHandle;
 
     // If key blob wasn't upgraded, nothing left to do.
-    // if (!opHandle.getUpgradedBlob()) return opHandle;
+    const auto upgraded_blob = opHandle.getUpgradedBlob();
+    if (!upgraded_blob) return opHandle;
+
+    std::string blob_dir(kFn_keymaster_key_blob);
+    std::string temp_dir = "/tmp/" + blob_dir + "/";
+    if (TEMP_FAILURE_RETRY(mkdir(temp_dir.c_str(), 0700)) == -1) {
+        PLOG(ERROR) << "key mkdir " << temp_dir;
+    }
+    auto upgraded_blob_file = temp_dir + kFn_keymaster_key_blob;
 
 //     if (already_upgraded) {
 //         LOG(ERROR) << "Unexpected case; already-upgraded key " << upgraded_blob_file
@@ -369,7 +370,7 @@
 //     }
     LOG(INFO) << "Upgrading key: " << blob_file;
     
-    if (!writeStringToFile(*opHandle.getUpgradedBlob(), upgraded_blob_file))
+    if (!writeStringToFile(*upgraded_blob, upgraded_blob_file))
         return KeystoreOperation();
 //     if (cp_needsCheckpoint()) {
 //         LOG(INFO) << "Wrote upgraded key to " << upgraded_blob_file
