--- a/MetadataCrypt.cpp
+++ b/MetadataCrypt.cpp
@@ -437,7 +437,10 @@
     }
 
     LOG(INFO) << "Mounting metadata-encrypted filesystem:" << mount_point;
-    mount_via_fs_mgr(mount_point.c_str(), crypto_blkdev.c_str(), needs_encrypt);
+    if (!mount_via_fs_mgr(mount_point.c_str(), crypto_blkdev.c_str(), needs_encrypt)) {
+        LOG(ERROR) << "Failed to mount metadata-encrypted filesystem";
+        return false;
+    }
     android::base::SetProperty("ro.crypto.fs_crypto_blkdev", crypto_blkdev);
 
     // Record that there's at least one fstab entry with metadata encryption
